### PR TITLE
Add boolean to specify if enterprise  number (ABN, GST, SIRET, ...) is mandatory or not to generate invoices

### LINKED ISSUE #9315: Allow super admin to activate if "company number" is mandatory or not
## Description
**- As a:** super admin
**- On page:** `/admin/invoice_settings/edit`
**- I want to be able to do:** I want to be able to activate or deactivate the fact that "company number" is required to print invoices.

Mockup: 

Context : https://github.com/openfoodfoundation/wishlist/issues/217

## Acceptance Criteria & Tests

1. Login as superadmin
2.  Go to `/admin/invoice_settings/edit`
3.  See that the checkbox is checked by default
4. Try to print invoices for company with empty number. You should have this message popin up : 

5. Go back to  `/admin/invoice_settings/edit`
6. Tick the checkbox off
7. You should be able to print invoices for those enterprises.

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #9315

#### What should we test?
 - Visit `/admin/invoice_settings/edit`
 - `REQUIRE AN ABN NUMBER TO GENERATE AN INVOICE?` should be checked
Capture d’écran 2022-06-23 à 16 39 38
 - For an enterprise without any ABN number,
Capture d’écran 2022-06-23 à 16 39 55
 - ... it must be impossible to print/send invoice (blocked with a js alert): 
Capture d’écran 2022-06-23 à 16 41 42
 
- then, add a ABN number
- you should be able to generate (print/send) invoice

 - then, remove ABN number
 - uncheck `REQUIRE AN ABN NUMBER TO GENERATE AN INVOICE?` under `/admin/invoice_settings/edit`
 - you should be able to generate (print/send) invoice

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/helpers/spree/admin/orders_helper.rb [hunk 1]
```diff
@@ -78,13 +86,20 @@ def send_invoice_link_without_url
...
 
-      def print_invoice_link
+      def print_invoice_link_with_url
         { name: t(:print_invoice),
           url: spree.print_admin_order_path(@order),
           icon: 'icon-print',
           target: "_blank" }
       end
 
+      def print_invoice_link_without_url
```
Comment (mkllnk) [inline app/helpers/spree/admin/orders_helper.rb:96]:
```suggestion
      def notify_about_required_abn
```

**Review by mkllnk — CHANGES_REQUESTED** [review]
Nice!

I have only style requests... :smile:

File: app/controllers/admin/invoice_settings_controller.rb [hunk 2]
```diff
@@ -19,6 +19,7 @@ def preferences_params
         :enable_invoices?,
         :invoice_style2?,
         :enable_receipt_printing?,
+        :enable_mandatory_abn_number?,
```
[the remark this answers is not in the document]
Comment (jibees) [reply app/controllers/admin/invoice_settings_controller.rb:22]: Therefor, I'm wondering if we should use ABN or something else?
As I can see that it's translated to `NIF` (ca), `GST No.` (nz), `Company Number` (en), `SIRET` (fr), ...

Why not use _company number_? I think it's the most generic we have

[the remark this answers is not in the document]
Comment (mkllnk) [reply app/controllers/admin/invoice_settings_controller.rb:22]: In OFN, we use the term *enterprise*. So we could call it Enterprise Number.

**Review by mkllnk — APPROVED** [review]
That reads much better now. :smile:

**Comment by filipefurtad0** [comment]
Agree! Merging :+1:

**Comment by filipefurtad0** [comment]
Hey @jibees,

While testing, it seems I've missed these lines from your detailed testing steps:

> then, remove ABN number

> uncheck REQUIRE AN ABN NUMBER TO GENERATE AN INVOICE? under /admin/invoice_settings/edit

> you should be able to generate (print/send) invoice


Indeed, it becomes possible to print, but:
- it is not possible to send the invoice - the red banner displayed and the invoice is not sent

So, basically, even though the checkbox is not ticked it is **not** possible to send invoices, if the ABN is **not** filled in. This is unrelated to the second business number (as I've indicated on my testing notes) -> thanks for your heads up @drummer83 :pray: 

I'll open an issue for this.

Side note: Should we consider improving the specs, for the case in which ABN mandatory is set to false?

**Pull request merged**